uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/PublicHolidayLeaveRequestCreation.php

```diff
@@ -47,10 +47,8 @@ public function createForAbsenceType(AbsenceType $absenceType) {
    * @param \CRM_HRLeaveAndAbsences_BAO_PublicHoliday $publicHoliday
    */
   public function createForContact($contactID, PublicHoliday $publicHoliday) {
-    $absenceTypes = $this->getAbsenceTypesWherePublicHolidaysMustBeTakenAsLeave();
-    foreach($absenceTypes as $absenceType) {
-      $this->create($contactID, $absenceType, $publicHoliday);
-    }
+    $absenceType = AbsenceType::getOneWithMustTakePublicHolidayAsLeaveRequest();
```

Comment:
looking at it now , I think it is better to have a generic method to get absencType and based on a flag you get the one with "must take public holiday is leave" since it is a special leave type .